discovery.md

```diff
@@ -110,6 +117,76 @@ be considered an event subscriber and a target user of this API.
 This API is specified as a REST API with well defined entities and relationships
 between those entities.
 
+To help explain the data model, the following non-normative pseudo yaml shows
+its basic conceptual structure:
+
+( `*` means zero or more, `+` means one or more, `?` means optional)
+
+```
+- services: *
+  - service: [unique identifier for producer of the specified event types]
+    description: [human string] ?
+    uri: [URI reference for human documentation] ?
+    specversion: [ce-speciversion value] ? (required?)
+    subscriptionuri: [URI to which the Subscribe request will be sent]
+    subscriptionconfig: *
+    authscope: [string] ?  (explain)
+    protocols: +
+    - protocol: [string]
+    types: *
+    - type: [ce-type value]
```

Comment:
I still wonder, if the de-normalized form is the right one here. I certainly understand the simplicity when it's supposed to support UIs. It's also nice and simple, if you look for a type emitted by a single service. In scenarios where an event type is emitted by multiple services, this will make the development of tools more complex. In case you want to provide tooling to generate some event handling or creation code, you would then have to assume that an event type is always defined the same way, or do complex structure comparisons.

Reply:
can you provide an example of what you mean by "de-normalized" in this context?

Reply:
In the proposal the "type" is a child of the "service". In reality the CloudEvents model defines "types" as an independent entity. You can have multiple "sources" or "services" using the type. If you define type as a child of service, it can be defined differently for every service. 
If you allow the definition of types independently from sources/services, it is a tricky question which context attributes are described in type and which are defined as part of the source/service. I could imagine a model where a service references a type and makes definitions for some attributes more specific. E.g. for a type it could be defined that it works with dataschema 1 or 2, and a service referring to this type could then restrict it to dataschema 1. I know this sounds really complex. I guess we need to agree on what kind of scenarios we want to support.

Reply:
perhaps open an issue so we can discuss it?

It kind of sounds like you're talking about enabling something like a type-ref thing, where the type is defined outside of a service but then referenced from within a service.  It might be useful to reduce the size of the response payload, but I kind of wonder if a user will need to know about that relationship.